package.json

```diff
@@ -66,9 +66,14 @@
     "node": ">=0.10.0"
   },
   "scripts": {
-    "test": "testem ci",
-    "lint-dts": "tslint sweetalert2.d.ts --format verbose",
-    "check-dts": "tsc sweetalert2.d.ts && npm run lint-dts",
+    "start": "gulp watch",
+    "fix:lint": "standard --fix",
+    "test": "npm run build && npm run check",
```

Comment:
make sure the build is fresh before running all required checks

Reply:
`build` is `build`, `test` is `test`, let's not mix concepts.

Reply:
In order to test properly, first you need to build. Building is a part of testing. Without this, tests could fail when the source code is fine, or tests could pass when the source code is broken.

Reply:
I.e. we don't have to depend on remembering to run `npm run build && npm test` (common script) rather than just `npm test`

Reply:
I understand your point, but I'm still against mixing the concepts. That was your issue recently, but that's because you didn't follow the contributing guide ;)

Take a look at the react configuration: https://github.com/facebook/react/blob/master/package.json#L102

As you can see the `test` task doesn't contain building, most of the developers don't expect `build` in the scope of the `test` task, by adding it we will bring more confusions than help.

Reply:
> I understand your point, but I'm still against mixing the concepts.

I don't think you do. My point is that they are two distinct yet related concepts. If you see my point that "Building is a part of testing" (in the context of this project) then why would the build script not be part of the test script?

> Take a look at the react configuration

Ok. I did. But not every project has to have the same setup as react.

The next project I checked (Flow, another by facebook) *does* include building as part of their package `test` script.. See [here](https://github.com/facebook/flow/blob/b56efe76adb93f7b53bd55ea44a10ad94dd02ee8/package.json#L47-L49) and [here](https://github.com/facebook/flow/blob/b56efe76adb93f7b53bd55ea44a10ad94dd02ee8/Makefile#L317-L319).

Also, Jest, a modern test framework with considerations for js build systems, will handle building for you when you run `jest`. See [this section of it's docs](https://github.com/facebook/jest#additional-configuration).

Let's focus on reasoning, instead of [appeal to authority](https://en.wikipedia.org/wiki/Argument_from_authority): The reason a test procedure should build first is because otherwise we are not testing the source, we are just testing whatever happens to be in the dist folder at the time. 

> That was your issue recently, but that's because you didn't follow the contributing guide ;)

If you are referring to PR #816 

1. This is like if I went into the shower, used some shampoo, my hair started falling out, I then read the instructions that say "apply shampoo to **dry** hair", and the shampoo company says that the reason my hair is falling out is because I didn't read the instructions. People do not read instructions for something that is (or rather, should be) as simple as shampoo.
2. I actually had run the `gulp watch` task, but it didn't work.. See PR #845

> most of the developers don't expect build in the scope of the test task, by adding it we will bring more confusions than help

`npm it` (equivalent to `npm install && npm test`) should work, unless the tests are broken. Like I said.. "Without this, tests could fail when the source code is fine, or tests could pass when the source code is broken." .. and when this happens, it is confusing. What possible confusion could come from including the building in the testing?

But this is not worth wasting any more time & energy arguing about. If you still prefer the build script to not be part of the test script, say the word and I'll undo this part of the PR.

Reply:
I love the shampoo example!

Apologies for being dummy :disappointed: Sometimes it takes a lot of time for me to understand valid arguments. Sometimes even [more than two years](https://github.com/sweetalert2/sweetalert2/issues/25).

Reply:
No problem, I am a dummy too sometimes lol ;)